### DISCUSSION TITLE
Move skip build check to javascript

### DISCUSSION THREAD

**Discussion by StefSchenkelaars** [ElMassimo/vite_ruby#172] [Ideas]
I really like the functionality of this gem to [prevent a build if nothing has changed](https://github.com/ElMassimo/vite_ruby/blob/b070774ea4e149b27ae0222e72fc5af3e5271bbb/vite_ruby/lib/vite_ruby/builder.rb#L16). I now used vite in another project (without ruby) where the build is performed every time, even if nothing has changed on the front-end part. 

So what about moving this functionality to a vite / rollup plugin? Would that also make sense to you?

**Comment by ElMassimo** [comment]
Hi Stef!

I'm not sure the benefits are worth it for most Vite apps:
- If no changes occurred, there's no need to redeploy them, [example](https://github.com/ElMassimo/vite_ruby/blob/main/netlify.toml#L6)
- If changes occurred, it's probably necessary to rebuild them

For Vite Ruby the logic is in Ruby because it's typically faster than starting a node process to perform the same check, so it wouldn't benefit from moving the logic to a Vite plugin. It also makes it [easier to test](https://github.com/ElMassimo/vite_ruby/blob/main/test/builder_test.rb#L34-L59).

---

Feel free to share what use cases you have in mind. I'm not sure Vite provides a good way to halt the build, so this script would probably not be implemented as a plugin alone, and might require an executable that triggers the build as needed.

As a possible alternative, in [îles](https://iles-docs.netlify.app/) I'm using [Nx Cloud](https://nx.app/), which can cache the results of any scripts, such as the `build` command. It might provide the benefit of skipping rebuilds and has the benefit of a shared cache between your local and the CI.
